### PR TITLE
feat(Chip): Add to prop for React Router

### PR DESCRIPTION
**Pull request by dLars99**
Adds a 'to' prop to the Chip component, which wraps the returned component in a React Router Link.

Also added the React Router as a decorator to Storybook, allowing Links in this and other components to render without creating an error.

### CODE REVIEW

**Review by erwstout — CHANGES_REQUESTED** [review]
Requesting changes based on slack convo about using `Link` as the element and `to` as an optional prop (as well as `href` optional)

**Comment by dLars99** [comment]
Added the JSXElementContstructor type as an option for the `element` prop. This will allow the user to pass in the Link (or another) component as `element={Link}`. I toyed with keeping it a simple string, but that was more limiting and required additional logic to parse.

File: packages/core/src/Chip/index.tsx [hunk 1]
```diff
@@ -152,6 +129,7 @@ const Chip: React.FC<ChipProps> = ({
       onClick={onClick}
       style={style}
       href={href}
+      to={to || href}
```
Comment (erwstout) [inline packages/core/src/Chip/index.tsx:132]: Should this be `to || null`? If `href` is present wouldn't it be applied to `to` no matter what? So then even `<a` or `<button` would have the `to` prop?

**Review by erwstout — CHANGES_REQUESTED** [review]

File: packages/core/src/Chip/index.tsx (hunk 1 above)
Comment (dLars99) [reply packages/core/src/Chip/index.tsx:132]: Fair point...was thinking flexibility, but that could cause issues

Comment (dLars99) [reply packages/core/src/Chip/index.tsx]: Changed to just {to}

**Review by erwstout — APPROVED** [review]

**Review by wliederer — APPROVED** [review]

**Pull request merged**